.github/actions/verify-metrics-snapshot/action.yaml

```diff
@@ -47,24 +47,25 @@ runs:
         restore-keys: |
           ${{ inputs.artifact_key }}
 
-    - name: Calculate diff between the snapshots 
+    - name: Calculate diff between the snapshots
       id: compare-snapshots
       if: ${{ (github.ref_name != 'main') && (steps.download-release-snapshot.outputs.cache-matched-key != '')  }}
+      continue-on-error: true
       shell: bash
       run: |
         python3 -m pip install prometheus-client
         python3 ./scripts/e2e/compare_metrics.py --file1 ./.metrics/${{ inputs.snapshot }}.txt --file2 ./.metrics/baseline_${{ inputs.snapshot }}.txt --output ./.metrics/diff_${{ inputs.snapshot }}.txt
         if [ $? -eq 1 ]; then
         echo "🛑 Differences found in metrics"
+        echo "has_diff=true" >> $GITHUB_OUTPUT
         exit 1
```

Comment:
do you still need this since you now have a cumulative step in .github/workflows/ci-e2e-all.yml? Why didn't this fail the workflows before?